.github/workflows/lint.yaml

```diff
@@ -19,16 +15,15 @@ jobs:
       - uses: actions/checkout@v2
 
       - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
           extra-packages: |
-            any::gh
-            any::purrr
-          needs: check
-
-      - name: Install package
-        run: R CMD INSTALL .
+            r-lib/lintr
```

Comment:
This is the only deviation from what's in the standard [example workflow](https://github.com/r-lib/actions/blob/v2-branch/examples/lint.yaml) for linting. 

I think we should always be linting with the latest GitHub version of `{lintr}`.